### PR TITLE
New Entry: Adding Counting Sort to Algorithm Terms

### LINKED ISSUE #4144: [Entry] Algorithms: Adding Counting Sort
### Reason/inspiration (optional)

In the Algorithms Folder, I would like to add the following Sorting Algo:
- Counting Sort

Within this doc, there should be 
- A brief explanation of the sorting algorithm (including time complexity):
- graphical/photographic representation 
- example code of sorting algo
- explanation of the sorting code 

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [x] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by JustinhSE**
### Description

Added Counting Sort as a Term in the Algos Folder. I included a conceptual and implementation explanation of the Sorting Algorithm. As well I also added the explanation for the time complexity. I wrote this issue as I learned this in my DSA class, and the professor simplified it a lot more than most online sources I used. I just wanted to share it with your community. 

Closes #4144 
### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [x] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [ ] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] Under "Development" on the right, I have linked any issues that are relevant to this PR (write "Closes #<issue number> in the "Description" above).

### CODE REVIEW

**Comment by JustinhSE** [comment]
@Sriparno08 do you mind explaining the errors or what I would need to fix in this pull request?

**Comment by dakshdeepHERE** [comment]
@JustinhSE don't worry about it. We will fix it in the end of the 2nd review.

**Comment by Sriparno08** [comment]
Hey @JustinhSE, thanks for contributing to Codecademy!

Before we move on to reviewing your entry, please delete the `vcs.xml` and `workspace.xml` files and commit the changes as they are not required for this PR.

**Comment by JustinhSE** [comment]
@Sriparno08 thanks, when i was working on it in VS it added that.

**Comment by JustinhSE** [comment]
@dakshdeepHERE alright thanks! I just finished removing the other xml files.

**Comment by JustinhSE** [comment]
I started working on another issue that i was assigned, merge sort, & it commited it for some reason? hoping that is not causing an error.

**Comment by dakshdeepHERE** [comment]
@JustinhSE it happens and that is sometimes a glitch from Github cli. Just go to your entry make a small change save and again push the change.

**Comment by JustinhSE** [comment]
alright thanks, I believe I did in my latest commit

**Comment by JustinhSE** [comment]
everything looks good on my side in the `files changed` section

**Review by Sriparno08 — COMMENTED** [review]
Hey @JustinhSE, thanks for contributing to Codecademy!

I've listed all the modifications that you need to make in your article below:

**Comment by JustinhSE** [comment]
@Sriparno08 just finished!

**Review by Sriparno08 — APPROVED** [review]
Good job, @JustinhSE 👍

This looks good for a second review.

File: content/general/concepts/algorithm/terms/counting-sort/counting-sort.md [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+---
+
+**Counting sort** is an out-of-place, non-comparison sorting [algorithm](https://www.codecademy.com/resources/docs/general/algorithm) that sorts a list with duplicate values efficiently. It is a helpful algorithm for sorting [arrays](https://www.codecademy.com/resources/docs/java/arrays) of small, non-negative integers, where it analyzes elements' frequency distribution and places them in their final sorted positions.
+
+## Explanation
+
+1. Find the maximum value:
+
+- Iterate through the input array `data` to find the largest element `max`.
+  - This determines the range of values we need to count.
```
Comment (dakshdeepHERE) [inline content/general/concepts/algorithm/terms/counting-sort/counting-sort.md:24]: Avoid using first- and second-person pronouns (e.g. "I", "we", "you") if possible. Please refactor this.

File: content/general/concepts/algorithm/terms/counting-sort/counting-sort.md [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+
+2. Create the count array:
+
+- Initialize a new array `count` with a `max + 1` size to hold each value's frequency. 
+    - The size is `max + 1` so that we can include the `max` element.
```
Comment (dakshdeepHERE) [inline content/general/concepts/algorithm/terms/counting-sort/counting-sort.md:29]: Avoid using first- and second-person pronouns (e.g. "I", "we", "you") if possible. Please refactor this.

File: content/general/concepts/algorithm/terms/counting-sort/counting-sort.md [hunk 3]
```diff
@@ -0,0 +1,121 @@
...
+
+- Loop through the `data` array again.
+- For each element `data[i]`, increment the corresponding count in the `count` array (`count[data[i]]++`). 
+    - This is to document the frequency distribution of elements in `data`.
+
+4. Create the start array:
+
+- Initialize a new array `start` of the same size as `count`.
+- Set `start[0]` to _0_.
+- Based on the frequency distribution found in `count`, we can find the starting indexes and store them in the `start` array (`start[j] = start[j-1] + count[j-1]` for `j=1` to `max`).
```
Comment (dakshdeepHERE) [inline content/general/concepts/algorithm/terms/counting-sort/counting-sort.md:42]: Avoid using first- and second-person pronouns (e.g. "I", "we", "you") if possible. Please refactor this.

File: content/general/concepts/algorithm/terms/counting-sort/counting-sort.md [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+   - Retrieve its corresponding position to the `temp` [variable](https://www.codecademy.com/resources/docs/java/variables) from the `start` array (`temp = start[data[a]]`).
+   - Place the element in the `result` array at that position (`result[temp] = data[a]`).
+   - Increment the `start` value for the next element with the same value (`start[data[a]]++`).
+
+6. Return the sorted array:
+
+- The `result` array now contains the elements of the original `data` array in sorted order.
+- Return the `result` array.
+
+## Implemented Example
```
Comment (dakshdeepHERE) [inline content/general/concepts/algorithm/terms/counting-sort/counting-sort.md:57]:
```suggestion
## Implementation
```
Lets call this header as implementation

File: content/general/concepts/algorithm/terms/counting-sort/counting-sort.md [hunk 5]
```diff
@@ -0,0 +1,121 @@
...
+
+The following example written in [Java](https://www.codecademy.com/resources/docs/java) shows an implementation of counting sort:
```
Comment (dakshdeepHERE) [inline content/general/concepts/algorithm/terms/counting-sort/counting-sort.md:59]:
```suggestion
The following example written in [Java](https://www.codecademy.com/resources/docs/java) shows an implementation of **counting sort**:
```

**Review by dakshdeepHERE — CHANGES_REQUESTED** [review]
I am requesting few minor changes from my side @JustinhSE. Otherwise this look good to merge.

File: content/general/concepts/algorithm/terms/counting-sort/counting-sort.md (hunk 3 above)
Comment (JustinhSE) [reply content/general/concepts/algorithm/terms/counting-sort/counting-sort.md]: just removed the first person pronouns, thanks for the heads up

File: content/general/concepts/algorithm/terms/counting-sort/counting-sort.md (hunk 2 above)
Comment (JustinhSE) [reply content/general/concepts/algorithm/terms/counting-sort/counting-sort.md]: just removed the first person pronouns, thanks for the heads up

File: content/general/concepts/algorithm/terms/counting-sort/counting-sort.md (hunk 1 above)
Comment (JustinhSE) [reply content/general/concepts/algorithm/terms/counting-sort/counting-sort.md]: just removed the first person pronouns, thanks for the heads up

**Comment by JustinhSE** [comment]
@dakshdeepHERE Finished!

**Review by dakshdeepHERE — APPROVED** [review]
LGTM 🚀

**Pull request merged**